scale/recipe/instance/recipe.py

```diff
@@ -50,3 +50,22 @@ def get_jobs_to_update(self):
             node.get_jobs_to_update(pending_job_ids, blocked_job_ids)
 
         return {'BLOCKED': blocked_job_ids, 'PENDING': pending_job_ids}
+
+    def get_original_leaf_nodes(self):
+        """Returns a mapping of non-superseded job and recipe IDs
+
+        :returns: A dict with node mapping of recipe leafs
+        :rtype: dict
+        """
+
+        leaf_job_ids = []
+        leaf_recipe_ids = []
+
+        for _, node in self.graph.items():
+            if node.is_original:
+                if node.node_type == JobNodeDefinition.NODE_TYPE:
+                    leaf_job_ids.append(node.job.id)
+                elif node.node_type == RecipeNodeDefinition.NODE_TYPE:
+                    leaf_recipe_ids.append(node.recipe.id)
+
+        return {'jobs': leaf_job_ids, 'recipes': leaf_recipe_ids}
```

Comment:
I would recommend returning a dict where the original leaf node names map to the instance node. The function will be more general and reusable that way. Also you need a check in here for the node's children to be empty, otherwise it is not limited to leaf nodes. Also if you are just iterating over the node instances (and not using node name) you can use `self.graph.values()` instead of `self.graph.items()`.